### PR TITLE
Scene Snapshot and State Refactor

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

Splits Scene state into two states:
- SceneState tracks metadata, extraneous state for scene loaded, modified etc
- SceneSnapshotState tracks event sourced snapshots, upon changing a snapshot applies it to the SceneState source of truth
- removes activeScene, specifying it explicitly based on context (studio sceneID or current location sceneID)

## References
closes #_insert number here_

### CODE REVIEW

File: packages/engine/src/scene/functions/createSceneEntity.ts
Comment (dinomut1) [inline packages/engine/src/scene/functions/createSceneEntity.ts:1]: We shouldn't delete this function, it's a very useful helper

**Review by dinomut1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/components/assets/ModelCompressionPanel.tsx
Comment (dinomut1) [inline packages/editor/src/components/assets/ModelCompressionPanel.tsx:1]: We shouldn't be defining a createTempEntity function within this UI panel file. CreateSceneEntity is a more flexible general use helper function that was sufficient here and has many potential uses elsewhere in the codebase

**Review by dinomut1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/components/assets/ModelCompressionPanel.tsx
Comment (HexaField) [reply packages/editor/src/components/assets/ModelCompressionPanel.tsx:1]: we don't have another use case for this yet. we can already do effectively the same thing by passing json and an empty object into generateEntityFromEntityJSON

Comment (dinomut1) [reply packages/editor/src/components/assets/ModelCompressionPanel.tsx:1]: This is a very useful function for proceduralism in general. There is no reason to remove it and it is not part of the scope of this PR anyways

**Pull request merged**